### DISCUSSION TITLE
Sequence Model Recommendation with YOLOv8

### DISCUSSION THREAD

**Discussion by aemiio** [ultralytics/ultralytics#16936] [Q&A] [answered]
Hello!

I’m working on a project involving Braille cell recognition using YOLOv8. However, due to the context dependence in Braille, I need to apply specific rules to interpret sequences of cells correctly.

For example:
- A letter is uppercase if preceded by a capital sign.
- A digit is recognized if preceded by a number sign.
- Braille contractions involve one or two cells, which can alter the meaning of the sequence.

I’m planning to use a sequence model to handle this context-sensitive interpretation of Braille cells. Could you recommend the best sequence model to pair with YOLOv8 for this type of task? Some options I’m considering are LSTM, GRU, Bidirectional LSTM, or Transformers.

Any insights or suggestions on which model would work best with YOLOv8 for handling sequential dependencies in Braille would be greatly appreciated!

Thank you in advance!

**Comment by glenn-jocher** [comment, marked as answer]
For handling sequential dependencies in Braille with YOLOv8, a Transformer model is often a strong choice due to its ability to capture long-range dependencies effectively. However, if computational resources are a concern, LSTM or Bidirectional LSTM can also be effective for context-sensitive tasks. Consider experimenting with these models to see which best suits your project's needs.

**Reply by aemiio** [reply]
thank you!!

**Reply by glenn-jocher** [reply]
You're welcome! If you have any more questions, feel free to ask.
